components/api/src/main/java/com/hotels/styx/api/LiveHttpMessage.java

```diff
@@ -100,14 +105,43 @@ default boolean chunked() {
      * all the traffic. This has the benefit of keeping the underlying TCP connection
      * open for connection pooling.
      * <p>
+     *
+     * The consumption begins immediately, running in the background.
      */
-    default void consume() {
-        body().drop().aggregate(1)
-                .thenApply(buffer -> {
-                    buffer.delegate().release();
-                    return null;
-                });
+    default void consumeInBackground() {
```

Comment:
I was hoping to get rid of this method. It is susceptible for denial of service: It keeps consuming as long as remote client keeps sending traffic.

Also this particular implementation looks to leak memory, as objects are not down referenced at `onNext`.
I'm sure this is not the case as you have deliberately left a comment (`// ignore`), but looking at this I'm still clueless as how it is supposed to work?

Reply:
Please note that the objects from `consume()` will be requests or responses that have already had the body consumed, so there is no further down-referencing necessary at this point. The comment says `// ignore` because it is not necessary to do anything with the request/response when called from this method.

I also wanted to get rid of it. However, I got a test failure in `HealthChecks.kt` when replacing ```                   .map {
                        it.consumeInBackground()
                        it.status().code() < 400
                    }``` 
with
```.flatMap {
  it.consume()
}
.map {
  it.status().code() < 400
}```